src/firmware/app/control/trajectory_planner.c

```diff
@@ -0,0 +1,254 @@
+#include "firmware/app/control/trajectory_planner.h"
+
+#include "firmware/shared/math/polynomial_1d.h"
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+#include "math.h"
+#include "stdio.h"
+
+void generate_constant_arc_length_segmentation(
+    FirmwareRobotPathParameters_t path_parameters, Trajectory_t* trajectory)
+{
+    TrajectoryElement_t reverse_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    double max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    double velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+
+    // Set the internal path parameter variables
+    TrajectoryElement_t* traj_elements = trajectory->trajectory_elements;
+
+    double final_speed                      = path_parameters.final_speed;
+    double initial_speed                    = path_parameters.initial_speed;
+    const double max_allowable_acceleration = path_parameters.max_allowable_acceleration;
+    const double max_allowable_speed        = path_parameters.max_allowable_speed;
+    const unsigned int num_segments         = path_parameters.num_segments;
+    const Polynomial2dOrder3_t path         = path_parameters.path;
+    double t_end                            = path_parameters.t_end;
+    double t_start                          = path_parameters.t_start;
+
+    trajectory->num_elements = path_parameters.num_segments;
+
+    // Variable used to flag if the path is moving "backwards" along the input path
+    unsigned int reverse_parameterization = 0;
```

Comment:
Could we use a boolean here instead? Just add `#include <stdbool.h>` at the top of the file to be able to use them.

Reply:
Changed to Bool